treq/client.py

```diff
@@ -195,6 +198,13 @@ def request(self, method, url, **kwargs):
                     b'content-type', [b'application/x-www-form-urlencoded'])
                 data = urlencode(data, doseq=True)
             bodyProducer = self._data_to_body_producer(data)
+        elif json:
```

Comment:
I think that this should be `elif json is not None` to avoid the odd asymmetry that passing `True` would generate a request with a body, while `False` would not.